File: rust_project/book-rust/src/ch4/rpn.py
stack = []

s = input('RPN:')

tokens = s.split()
for t in tokens:
    t = t.strip()
    try:
        stack.append(float(t))
    except ValueError:
        b = stack.pop()
        a = stack.pop()
        if t == "+": stack.append(a + b)
        elif t == "-": stack.append(a - b)
        elif t == "*": stack.append(a * b)
        elif t == "/": stack.append(a / b)
        else: raise Exception("undifined token:" + t)

print(stack.pop())

File: rust_project/book-rust/src/ch4/rpn.rs
use std::io;
fn main() {
    let mut stack: Vec<f64> = vec![];
    println("RPN:");
    let mut s = String::new();
    io::stdin().read_line(&mut s).expect("input error");
    let tokens = s.split_whitespace();
    for tok in tokens {
        let t = tok.trim();
        match t.parse<f64>() {
            Ok(v) => { stack.push(v); continue; },
            Err(_) => 0.0,
        };

        let b = stack.pop().unwrap();
        let a = stack.pop().unwrap();
        match t {
            "+" => stack.push(a + b),
            "-" => stack.push(a - b),
            "*" => stack.push(a * b),
            "/" => stack.push(a / b),
            _ => panic!("undifiend token:{}", t),
        }
    }
    println!("{}", stack.pop().unwrap());
}
